### ISSUE TITLE
Exposing JSon interface of the magenement server

### ISSUE THREAD

**Issue by sskhiri** [roq-messaging/RoQ#58]
# Sub-task of the issue #56

The objective is to expose a clear interface non-java dependent to any management console through a broadcast mechanism. 
For this task we are considering using 
- BSON http://bsonspec.org/
- Hot Rod https://community.jboss.org/wiki/HotRodProtocol
- Thrift thrift.apache.org

**Comment by sskhiri** [comment]
Using the mongodb-driver (https://github.com/mongodb/mongo-java-driver/) for BSON, I got those two JSON output can you check if they are OK ?

## List of hosts

{ "hosts" : [ "127.0.1.1" , "127.0.1.2" , "127.0.1.3" , "127.0.1.4"]}

## List of Queue states

{ "Queues" : [ { "Name" : "queue1" , "Host" : "127.0.1.1" , "State" : false} , { "Name" : "queue2" , "Host" : "127.0.1.2" , "State" : false} , { "Name" : "queue3" , "Host" : "127.0.1.3" , "State" : false}]}

**Comment by sskhiri** [comment]
## Update and storage

The management controller store the on-line configuration received by the the Global configuration manager in SQL lite storage. According to the updates, it infers the delta and update the queues state. Notice that currently no queues are removed as they are not present any more in the live configuration they are updated as "OFF".

## The Broadcasted information

From the GlobalConfiguration server :5004, any client can subscribe and received every min a configuration in BSON. As mentioned, we use the mongoDB java driver for BSON. Example of encode/decode can be found in  org.roqmessaging.management.server.UnitTestManagement.java 
Basically, we send a multi message envelope with ZMQ, then the message is composed by 3 parts:
- The CMD_ID: 1500 for MNGT_UPDATE_CONFIG as defined in RoQConstants
- The Queues: the list of Queue states encoded in BSON { "Queues" : [ { "Name" : "queue1" , "Host" : "127.0.1.1" , "State" : false} , { "Name" : "queue2" , "Host" : "127.0.1.2" , "State" : false} , { "Name" : "queue3" , "Host" : "127.0.1.3" , "State" : false}]}
- The Hosts: the list of host addresses encoded in BSON { "hosts" : [ "127.0.1.1" , "127.0.1.2" , "127.0.1.3" , "127.0.1.4"]}

## Future works

Next steps implementing the request interface of the management console.
